lib/daterangepicker/daterangepicker.js

```diff
@@ -240,7 +240,7 @@
                 silent = !!options.silent;
 
             if(month !== monthToDisplay.month() || year !== monthToDisplay.year()){
-                this.showMonth();
+                this.showMonth(year, month);
```

Comment:
I would suggest you don't pass the args here, but rather do this:

```
this.monthToDisplay.set({
    month: month,
    year: year
});
this.showMonth();
```

The `this.monthToDisplay` property is a "model" for the month that is currently displayed, and `this.showMonth` always relies on the fact that `this.monthToDisplay` is up-to-date.